### PR TITLE
add http basic auth if password in connection string

### PR DESCRIPTION
**Pull request by devinstevenson**
@jingw small PR. Thanks.

CLA submitted.
Enables the use of providing credentials to Presto server via BasicAuth

add password keyword arg to Cursor
If password provided add AUTHORIZATION header to request via requests.HTTPBasicAuth
force HTTPS protocol when using password with BasicAuth

### CODE REVIEW

File: pyhive/presto.py [hunk 1]
```diff
@@ -104,7 +108,10 @@ def __init__(self, host, port='8080', username=None, catalog='hive', schema='def
         if protocol not in ('http', 'https'):
             raise ValueError("Protocol must be http/https, was {!r}".format(protocol))
         self._protocol = protocol
-
+        self._auth = None
+        if password:
+            self._auth = HTTPBasicAuth(username, self._password)
+            self._protocol = 'https'  # require https when authenticating
```
Comment (jingw) [inline pyhive/presto.py]: Can you require this explicitly like https://github.com/dropbox/PyHive/blob/master/pyhive/hive.py#L88 and https://github.com/dropbox/PyHive/blob/master/pyhive/tests/test_hive.py#L175?

Someone might pass protocol=http and pass a password, and the protocol would get silently ignored.

File: pyhive/presto.py [hunk 2]
```diff
@@ -104,7 +108,10 @@ def __init__(self, host, port='8080', username=None, catalog='hive', schema='def
         if protocol not in ('http', 'https'):
             raise ValueError("Protocol must be http/https, was {!r}".format(protocol))
         self._protocol = protocol
-
+        self._auth = None
```
Comment (jingw) [inline pyhive/presto.py]: nitpick: move to else branch of if password, so we're not setting it and mutating it

File: pyhive/presto.py [hunk 3]
```diff
@@ -104,7 +108,10 @@ def __init__(self, host, port='8080', username=None, catalog='hive', schema='def
...
+        if password:
```
Comment (jingw) [inline pyhive/presto.py]: password='' is technically valid, so this should be based on `password is None`

File: pyhive/presto.py [hunk 4]
```diff
@@ -76,12 +77,14 @@ class Cursor(common.DBAPICursor):
     visible by other cursors or connections.
     """
 
-    def __init__(self, host, port='8080', username=None, catalog='hive', schema='default',
-                 poll_interval=1, source='pyhive', session_props=None, protocol='http'):
+    def __init__(self, host, port='8080', username=None, password=None, catalog='hive',
```
Comment (jingw) [inline pyhive/presto.py]: password should be at end in case people are relying on argument order

**Comment by devinstevenson** [comment]
@jingw  Thanks for the comments. Changes made and added a test.

**Comment by jingw** [comment]
Looks like test is fixable by calling .cursor() on the connection. This is a quirk of the Connection class not actually doing anything and punting all the work to Cursor. (arguably a bug but oh well...)

**Comment by kruppel** [comment]
@jingw Would it be possible to cut a new patch release with this change? Thanks!

**Pull request merged**